### ISSUE TITLE
Heading level offset

### ISSUE THREAD

**Issue by jbenet** [gohugoio/hugo#6373]
Many markup processors, [including blackfriday](https://github.com/russross/blackfriday/blob/v2.0.1/html.go#L91-L94), support offsetting heading levels. this is key when composing content, say into a single big page, or a few pages.

There's [a hacky way](https://discourse.gohugo.io/t/option-to-shift-headings/6136) of doing this in the forums, but this breaks shortcode usage, and a bunch of other things.

This can be solved by just being able to set:
```
blackfriday.HTMLRendererParameters.HeadingLevelOffset = level
```

This can't be a config-wide thing though (as many of the other blackfriday parameters are). It has to be set per render call, and likely cannot be set in the page itself (ie the source page may not know the level of header it should offset to, only its caller may).

So, basically, when calling something like the following:
```
{{ $b := $a.GetPage foo }}
{{ $b.Content }}
```

It should be possible to set params to blackfriday for the that page's render:

```
{{ $b := $a.GetPage foo (dict "config" blackfridayCfg) }}
{{ $b.Content }}
```
or even something like this (change it every time)
```
{{ $prev :=  .Config.HeadingLevelOffset }}
{{ .Config.HeadingLevelOffset = $level }}
{{ $b := $a.GetPage foo }}
{{ $b.Content }}
{{ .Config.HeadingLevelOffset = $prev }}
```

**Comment by computator** [comment]
This is something that would still be useful. I made a custom partial that uses some regex substitutions to get around this but native support would be much better.

**Comment by computator** [comment]
Bump again

**Comment by computator** [comment]
Not stale afaik
